Add a page on Godot system requirements

Calinou opened the pull request:
Let me know if the format is good and I'll complete the TODO items :slightly_smiling_face:

I'm not sure if we should feature two separate sets of minimum/recommended specs for exported projects (one for simple projects, one for more complex projects), as a project's complexity is a subjective topic.

I also don't know if we should feature a 3rd tier such as "optimal" (e.g. for a smooth 4K 60 FPS experience). This something a lot of AAA games have been doing recently ([example](https://user-images.githubusercontent.com/180032/233505125-c7eb3053-ca9b-4d3e-9d2f-97373c527eeb.png)), but I feel it would introduce too many combinations to check and maintain.

This also adds a note about Windows 7/8/8.1 support being done on a best-effort basis. This closes https://github.com/godotengine/godot/issues/77304.

Once we agree on a final design, I can work on a 3.x version of this page.

- This closes https://github.com/godotengine/godot-docs/issues/6497.

## TODO

- [x] Add editor system requirements.
- [x] Add recommended requirements for exported projects.

bruvzg commented on about/system_requirements.rst:
```diff
@@ -0,0 +1,116 @@
+| **RAM**              | - **For native exports:** 2 GB                                                          |
+|                      | - **For web exports:** 4 GB                                                             |
++----------------------+-----------------------------------------------------------------------------------------+
+| **Storage**          | 100 MB (used for the executable, project files and cache)                               |
```
macOS release export executable is 130 MB (since it includes 2 architectures).

bruvzg commented on about/system_requirements.rst:
```diff
@@ -0,0 +1,116 @@
++----------------------+-----------------------------------------------------------------------------------------+
+| **Storage**          | 100 MB (used for the executable, project files and cache)                               |
++----------------------+-----------------------------------------------------------------------------------------+
+| **Operating system** | - **For native exports:** Windows 7, or Linux distribution released after 2016          |
```
macOS 10.12+ (for Compatibility renderer, theoretically I do not think anyone tested it), 10.15+ (for Vulkan, some limited functionality on 10.14 is possible).

bruvzg commented on about/system_requirements.rst:
```diff
@@ -0,0 +1,116 @@
+|                      |                                                                                         |
+|                      | - **iOS:** SoC with any 64-bit ARM CPU                                                  |
+|                      |                                                                                         |
+|                      |   - *Example: Apple A7 (iPhone 5S)*                                                     |
```
For Vulkan renderer it's likely A9+ (iPhone 6S / SE 1.gen.), A7 have Metal support, but not all features used by Godot are supported.

and-rad commented:
I think "optimal" is unneccessary. When going with minimum and recommended, it's probably a good idea to come to an agreement on what kinds of features these include. It might also be worth it to have minimum and recommended specs separated by 2D/3D because a simple 3D game can easily become way more demanding than a complex 2D game.

By "agreement" I'm thinking something like, what's the required spec to get a game running at 30 (60?) FPS with MSAA, SDFGI, SSAO enabled, a particle system with X amount of particles in the scene and between Y and Z vertices coming from N meshes, drawn with K number of draw calls? That would then count as the baseline for the "recommended" settings.

How to reach this kind of agreement, I have no idea 🤷‍♂️

lostminds commented:
There's a nice pr https://github.com/godotengine/godot/pull/77740 that will hopefully reduce iOS device requirements, so once that is merged I don't think an A9 processor will be required anymore for vulkan renderers on iOS.

Regarding minimum/recommended my perspective would be to give the minimum requirements the primary focus. As in what will be required to run the game at all (not get a black screen or crash on startup). And for that purpose also perhaps add some practical hints on how this can be configured on the different platforms to avoid distribution to users/devices that don't support your game. 

For example even though the actual requirement is typically specific graphics hardware, this can not be specified as a required capability in the mobile app stores (Apple App Store and Google Play), so it's not practically that useful. Instead you need to specify a higher OS version requirement that indirectly means the device will have the required hardware capabilities to support that OS. And getting a hint about this would I think be useful to users looking to distribute mobile  games.

Calinou commented:
I've added specifications for the editor and updated the page to mention ARM Linux support (with suggested ARM boards to run Godot on). This should be ready to merge now.

mhilbrunner commented:
Thank you! 🙏 Merged.

mhilbrunner commented:
Cherry-picked to 4.2.

aaronfranke commented:
> Once we agree on a final design, I can work on a 3.x version of this page.

That would be very nice to have documented.
